mlir-sycl/test/Transforms/constant-propagation.mlir

```diff
@@ -1,42 +1,1155 @@
+// CHECK-NEXT:        %[[VAL_138:.*]] = arith.constant 0 : index
+// CHECK-NEXT:        %[[VAL_139:.*]] = arith.constant 512 : index
+// CHECK-NEXT:        %[[VAL_140:.*]] = arith.constant 1 : index
+// CHECK-NEXT:        %[[VAL_141:.*]] = arith.constant 1 : index
+// CHECK-NEXT:        %[[VAL_142:.*]] = sycl.range.constructor(%[[VAL_139]], %[[VAL_140]], %[[VAL_141]]) : (index, index, index) -> memref<1x!sycl_range_3_>
+// CHECK-NEXT:        %[[VAL_143:.*]] = memref.load %[[VAL_142]]{{\[}}%[[VAL_138]]] : memref<1x!sycl_range_3_>
+// CHECK-NEXT:        %[[VAL_144:.*]] = arith.constant 0 : index
+// CHECK-NEXT:        %[[VAL_145:.*]] = sycl.global_offset : !sycl_id_3_
+// CHECK-NEXT:        memref.store %[[VAL_145]], %[[VAL_122]]{{\[}}%[[VAL_144]]] : memref<1x!sycl_id_3_>
+// CHECK-NEXT:        memref.store %[[VAL_131]], %[[VAL_123]]{{\[}}%[[VAL_144]]] : memref<1x!sycl_range_3_>
+// CHECK-NEXT:        memref.store %[[VAL_137]], %[[VAL_124]]{{\[}}%[[VAL_144]]] : memref<1x!sycl_range_3_>
+// CHECK-NEXT:        memref.store %[[VAL_143]], %[[VAL_125]]{{\[}}%[[VAL_144]]] : memref<1x!sycl_range_3_>
+// CHECK-NEXT:        return
+// CHECK-NEXT:      }
+  func.func @init.3(%offset: memref<1x!sycl_id_3_>,
+                      %global_size: memref<1x!sycl_range_3_>,
+                      %local_size: memref<1x!sycl_range_3_>,
+                      %num_work_groups: memref<1x!sycl_range_3_>) {
+    %c0 = arith.constant 0 : index
+    %offset.val = sycl.global_offset : !sycl_id_3_
+    %global_size.val = sycl.num_work_items : !sycl_range_3_
+    %local_size.val = sycl.work_group_size : !sycl_range_3_
+    %num_work_groups.val = sycl.num_work_groups : !sycl_range_3_
+    memref.store %offset.val, %offset[%c0] : memref<1x!sycl_id_3_>
+    memref.store %global_size.val, %global_size[%c0] : memref<1x!sycl_range_3_>
+    memref.store %local_size.val, %local_size[%c0] : memref<1x!sycl_range_3_>
+    memref.store %num_work_groups.val, %num_work_groups[%c0]
+        : memref<1x!sycl_range_3_>
+    func.return
+  }
+
+// CHECK-LABEL:     func.func @__impl.1_k0(
+// CHECK-SAME:                             %[[VAL_158:.*]]: memref<?xindex>, %[[VAL_159:.*]]: memref<?xindex>, %[[VAL_160:.*]]: memref<?xindex>, %[[VAL_161:.*]]: memref<?xindex>) {
+// CHECK-NEXT:        %[[VAL_162:.*]] = arith.constant 0 : index
+// CHECK-NEXT:        %[[VAL_163:.*]] = arith.constant 0 : i32
+// CHECK-NEXT:        %[[VAL_164:.*]] = memref.alloca() : memref<1x!sycl_id_1_>
+// CHECK-NEXT:        %[[VAL_165:.*]] = memref.alloca() : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_166:.*]] = memref.alloca() : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        %[[VAL_167:.*]] = memref.alloca() : memref<1x!sycl_range_1_>
+// CHECK-NEXT:        call @__init.1_k0_impl.1(%[[VAL_164]], %[[VAL_165]], %[[VAL_166]], %[[VAL_167]]) : (memref<1x!sycl_id_1_>, memref<1x!sycl_range_1_>, memref<1x!sycl_range_1_>, memref<1x!sycl_range_1_>) -> ()
```

Comment:
See how the function call is the only change here as we will not be using the original function anymore.